tickit/adapters/interpreters/command/command_interpreter.py

```diff
@@ -84,16 +93,33 @@ async def handle(
                 indicating whether an interrupt should be raised by the adapter.
         """
         for _, method in getmembers(adapter):
-            command = getattr(method, "__command__", None)
+            message = message
+            command = cast(Command, getattr(method, "__command__", None))
             if command is None:
                 continue
-            args = command.parse(message)
-            if args is None:
+            parse_result = command.parse(message)
+            if parse_result is None:
+                continue
+            (
+                match_groups,
+                _,
+                match_end,
+                message_end,
+            ) = parse_result
+            if match_end != message_end:
+                # We want the whole (formatted) message to match a command
                 continue
             args = (
                 argtype(arg)
-                for arg, argtype in zip(args, get_type_hints(method).values())
+                for arg, argtype in zip(match_groups, get_type_hints(method).values())
             )
+            # can we use signature instead to more cleverly do conversions? i.e. if
+            # type hints aren't available just pass on with no conversion? Otherwise
+            # we can get unhelpful error messages.
+            # Also, should we deal with str/bytes conversions seperately?
+            # Create a mapping function that does conversion rather than argtype(arg)?
+            # It would take args and inspect.Parameters as parameters?
+            # Is type hints enough? - doesn't know about non-hinted vars
```

Comment:
This is definitely worth investigating at some point. My current thoughts are:

- The case of no type hints should be accounted for, either by raising an error or dumping the raw `AnyStr` on the user
- Due to the complexity of `bytes<->string` it makes most sense to dump the raw `AnyStr` on the user and let them deal with it, the complexity of passing through encodings / decodings seems unnecessary when the user can do it in their endpoint.
- A mapping function would be nice if we could come up with one that satisfied users needs across the board, but I frankly don't see that happening in a way that's better than using the default casts of the types

Reply:
I think discussions like this are better had in new issues than in code comments/PRs. Can we remove the code in the comment and make a new issue?